### PR TITLE
Current structure of the recieved order object over webhooks

### PR DESCRIPTION
**Pull request by cactusaurelius**
## :memo: Description

The address in orders received over webhooks is named dilevery_address

### :gem: Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

### :scroll: Example code
```js
      order.shipping = order.shipping || order.deliveryAddress;

``` 

## :checkered_flag: Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code

### CODE REVIEW

File: src/order.ts [hunk 1]
```diff
@@ -4,6 +4,7 @@ export interface Order {
   id: string;
   items: Partial<OrderItem>[];
   shipping: Address;
+  deliveryAddress?: Address;
```
Comment (HazemKhaled) [inline src/order.ts:7]: this order object fro sales order, you need to create new PurchaseOrder type

**Review by HazemKhaled — CHANGES_REQUESTED** [review]

File: src/purchaseorder.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  subTotal: number;
+  isInclusiveTax: boolean;
+  items: item[];
+  note: string;
+  attention: string;
+  shipping: Address;
+  deliveryAddress: Address;
+}
+
+export interface item {
```
Comment (HazemKhaled) [inline src/purchaseorder.ts:24]: The naming convention should be camel case
use `PurchaseOrderItem` instead

make sure all fields are included https://docs.knawat.io/#tag/Orders/paths/~1async~1orders/post

File: src/purchaseorder.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  status: string;
+  currencyCode?: string;
+  currencySymbol?: string;
+  subTotal: number;
+  isInclusiveTax: boolean;
+  items: item[];
+  note: string;
+  attention: string;
+  shipping: Address;
+  deliveryAddress: Address;
```
Comment (HazemKhaled) [inline src/purchaseorder.ts:21]: Instead, add issue to @neer14 to fix deliveryAddress and remove it from here

in you app SEA, extend it for now

**Review by HazemKhaled — CHANGES_REQUESTED** [review]

File: src/purchaseorder.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  status: string;
+  currencyCode?: string;
+  currencySymbol?: string;
+  subTotal: number;
+  isInclusiveTax: boolean;
+  items: Item[];
+  note: string;
+  attention: string;
+  shipping: Address;
+}
```
Comment (neer14) [inline src/purchaseorder.ts:21]: Missing date,referenceNumber, receivedStatus, billedStatus, exchangeRate, isDropShipment, salesorderId, adjustment, adjustmentDescription, subTotalInclusiveOfTax,taxTotal, deliveryOrgAddressId, deliveryCustomerId, createdTime, lastModifiedTime, comments, bills, salesorders, taxes, purchasereceives

File: src/purchaseorder.ts [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  sku: string;
+  name: string;
+  rate: number;
+  quantity: number;
+  taxId: string;
+  taxName: string;
+  taxPercentage: number;
+  description: string;
+  discount: number;
+  total: number;
```
Comment (neer14) [inline src/purchaseorder.ts:33]: missing id,accountId,taxType

**Review by HazemKhaled — APPROVED** [review]

**Pull request merged**